contracts/profit/src/state/buy_back.rs

```diff
@@ -24,25 +28,41 @@ use crate::{
 
 use super::{idle::Idle, Config, ConfigManagement, SetupDexHandler, State, StateEnum};
 
-pub type BuyBackCurrencies = NonNativePaymentGroup;
-
 #[derive(Serialize, Deserialize)]
 pub(super) struct BuyBack {
-    contract_addr: Addr,
+    profit_contract: Addr,
     config: Config,
     account: Account,
-    coins: Vec<CoinDTO<BuyBackCurrencies>>,
+    coins: Vec<CoinDTO<PaymentGroup>>,
 }
 
 impl BuyBack {
+    /// Until [issue #7](https://github.com/nolus-protocol/nolus-money-market/issues/7)
+    /// is closed, best action is to verify the pinkie-promise
+    /// to not pass in [native currencies](Native) via a debug
+    /// assertion.
     pub fn new(
-        contract_addr: Addr,
+        profit_contract: Addr,
         config: Config,
         account: Account,
-        coins: Vec<CoinDTO<BuyBackCurrencies>>,
+        coins: Vec<CoinDTO<PaymentGroup>>,
     ) -> Self {
+        debug_assert!(
+            coins.iter().all(|coin_dto: &CoinDTO<PaymentGroup>| {
+                <
+                currency::non_native_payment::NonNativePaymentGroup as finance::currency::Group
```

Comment:
`NonNativePaymentGroup` is going to be removed ... migrate the code accordingly

Reply:
On a related note, it is not harmful to get some amount in Nls here. There is going to be an extra transfer out to the DEX and that is the only drawback. The SwapExactIn is going to manage its presence and filter it out. Finally, the TransferIn would bring it back to Nolus accumulated into the total amount.